### DISCUSSION TITLE
Rendering with GUI settings (distortion and glow mode)

### DISCUSSION THREAD

**Discussion by cvachha** [NVlabs/instant-ngp#740] [Q&A] [answered]
Hello,

I was wondering how to render scenes using run.py while keeping settings from the GUI such as lens distortion (such as latlong to render 360 images) or glow mode. I am trying to render 360 scenes, but calling run.py renders them without distortion. I am able to call run.py on the model, but the render does not show those additional settings/changes. Is there a way to specify those settings within run.py?

Thank you.

**Comment by Tom94** [comment, marked as answer]
Yes, you can change all of these settings through Python. The concrete bindings are listed in `python_api.cu`. Cheers!

**Reply by cvachha** [reply]
Thank you for the response. I looked through the python_api.cu and found that ECameraDistortionMode can have the value of the camera distortion, but how do I set it to LatLong from the python file? I also noticed that glow is found in the testbed_nerf.cu file but not in the python_api.cu. Should I set these values in the cuda files?

**Reply by Tom94** [reply]
Ah, I hadn't realized the glow modes didn't have bindings yet. I just pushed a new commit that includes them.

You can use `testbed.nerf.render_distortion.mode = ngp.CameraDistortionMode.LatLong` for the distortion.

**Reply by cvachha** [reply]
Thank you! I had to slightly edit the line you mentioned for distortion to `testbed.nerf.render_distortion.mode = ngp.ECameraDistortionMode.LatLong` which uses ECameraDistortionMode instead of CameraDistortionMode.

**Reply by T-Photoworks** [reply]
Sorry to join cvachha's question.
I am a 360VR videographer. I also wanted to use LatLong videos for my material. Then I found this topic. But I am a beginner in python and scripting. 
Do I enter this script when I transform the 360VR video I put in the script？Or when building?
Does it also input for instant-ngp's Dir? 
I hope you can answer my stupid question. Thanks.
